github/repos_contents.go

```diff
@@ -240,16 +240,27 @@ const (
 // or github.Zipball constant.
 //
 // GitHub API docs: https://developer.github.com/v3/repos/contents/#get-archive-link
-func (s *RepositoriesService) GetArchiveLink(ctx context.Context, owner, repo string, archiveformat archiveFormat, opt *RepositoryContentGetOptions) (*url.URL, *Response, error) {
+func (s *RepositoriesService) GetArchiveLink(ctx context.Context, owner, repo string, archiveformat archiveFormat, opt *RepositoryContentGetOptions, followRedirects bool) (*url.URL, *Response, error) {
 	u := fmt.Sprintf("repos/%s/%s/%s", owner, repo, archiveformat)
 	if opt != nil && opt.Ref != "" {
 		u += fmt.Sprintf("/%s", opt.Ref)
 	}
+	resp, err := s.getArchiveLinkFromURL(ctx, u, followRedirects)
+	resp.Body.Close()
```

Comment:
It looks to me like the body is being closed twice - once on line 280, then a second time on line 249.

(Additionally, `err` needs to be checked first before `resp.Body.Close()` because `resp` might be `nil`...
however, I believe the correct solution is to actually delete line 249 entirely, since it is already being handled.)